### DISCUSSION TITLE
Welcome to UniqueBible Discussions!

### DISCUSSION THREAD

**Discussion by eliranwong** [eliranwong/UniqueBible#89] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by darrelwright** [comment]
I am wanting to use UBA as my primary electronic bible. I come from having used theWord bible for over ten years. One of the main features I need a bible program to do is allow me to search for a Strongs number where is translated as a particular word. For example in theWord I enter the search phrase G746 TR αρχη and it returns 12 verses containing G746 translated as αρχη. How do I do that in UBA?

**Reply by otseng** [reply]
Could you create a new issue?
https://github.com/eliranwong/UniqueBible/issues

I would like to see a similar feature like you requested.

**Comment by darrelwright** [comment]
I just posted a bug report and sent the following to Eliran,

Just thought I'd let you know. I just reported a bug relating to the 
control panel. If you enable the control panel to start at startup and 
to replace the command line, things really don't play. If you turn those 
settings off then things work.:-)

The issue occurs in 21.39 when you enable control panel at startup and 
to replace the command line.

**Reply by eliranwong** [reply]
this issue was fixed.

**Comment by jdecent** [comment]
Can someone inform me on the source materials or history of the OHGB translation?

**Reply by eliranwong** [reply]
@jdecent 

Same as MOB, MIB, MAB, MPB, MTB, Hebrew and Greek data are my little integrated work from two projects.

For Hebrew portion, you may read:

https://github.com/eliranwong/OpenHebrewBible

For Greek portion, you may read:

https://github.com/eliranwong/OpenGNT

**Comment by cquek-ubs** [comment]
Hi, we are planning to create a version for use in China.  Is it possible to whitelabel a version based on your open source code?
Would you have the capacity to provide consulting or development support (with remuneration)?

**Comment by onegladiator4u** [comment]
Hey @eliranwong 
i have created Malayalam book names & abbreviations in this file 
[BibleBooks.zip](https://github.com/eliranwong/UniqueBible/files/9040642/BibleBooks.zip)
Can you add those information also to the `BibleBooks.py` in root and util folders ?

**Reply by otseng** [reply]
Done

**Reply by onegladiator4u** [reply]
> Done

now i just need to add my file like those sample files in language folder right ?

**Reply by onegladiator4u** [reply]
i got this error while updating . is this a problem ?
![image](https://user-images.githubusercontent.com/48617390/177583581-2166fe22-123a-4e64-8389-aceea64fbbf8.png)
got errors for those 2 files too in that folder

**Reply by onegladiator4u** [reply]
@otseng i think creating a folder is problematic . It didn't create a folder named language. It happened also with audio folder too. I don't know whether you remember it

**Reply by onegladiator4u** [reply]
i created a folder language and changed bible version file and updated again. its updated . Just wanted to let you know

**Reply by otseng** [reply]
My bad.  I always forget to add the folder to the patch file.

**Reply by onegladiator4u** [reply]
> My bad. I always forget to add the folder to the patch file.

@otseng bro i have problems starting uba after that update 
![image](https://user-images.githubusercontent.com/48617390/177594333-11bbd8c0-0ee3-48c4-9cca-4a128df73a22.png)

**Reply by otseng** [reply]
Can you try replacing BibleBooks.py with this..
[BibleBooks.zip](https://github.com/eliranwong/UniqueBible/files/9056645/BibleBooks.zip)
.

**Reply by onegladiator4u** [reply]
which file should I replace . root folder or util folder or both ? its in both the folders

**Reply by otseng** [reply]
Not sure why you have one in the root.  Delete that one.

It should only be in the util folder.  Replace that one.

**Reply by onegladiator4u** [reply]
> Not sure why you have one in the root. Delete that one.

looks like i may have many files that should have deleted. 


> It should only be in the util folder. Replace that one.

replaced it bro. its working now . Thank You so much .
just wanted to clear one doubt that we can't change this to book language right ?
![image](https://user-images.githubusercontent.com/48617390/177620990-14eada56-5e54-42f5-a0ec-d1cc9ff26774.png)

**Reply by otseng** [reply]
Do you have "Select Bible Abbreviation" set to "ML"? ...

![Screenshot from 2022-07-06 17-13-55](https://user-images.githubusercontent.com/806056/177644785-6ea694f5-2fd6-496e-a0d0-0fc9f8c114ef.png)

**Reply by onegladiator4u** [reply]
i coudn't see ml in that.

![image](https://user-images.githubusercontent.com/48617390/177708675-25935e66-50b8-4f6f-a75f-fe7fb99a35c7.png)
how can i add it?

**Reply by otseng** [reply]
Oops, I mentioned how to do it in the [MR](https://github.com/eliranwong/UniqueBible/pull/1105), but didn't mention it here.

Download the ml.biblebooks file in Resources/Add/GitHub Bible Abbreviations
![Screenshot from 2022-07-07 08-11-50](https://user-images.githubusercontent.com/806056/177770796-bf661d34-e382-4ab3-9c6d-c643130a2209.png)
ML should then show up to be selectable as Bible abbreviation.

**Reply by onegladiator4u** [reply]
@otseng I actually created ml.biblebooks, so couldn't see it in  Resources/Add/GitHub Bible Abbreviations.
So i moved the file and downloaded again. It downloaded. 
But didn't show up in Abbreviations list. i restarted app too. what might be the problem ?


[ml.zip](https://github.com/eliranwong/UniqueBible/files/9063637/ml.zip)
Also Can you update this one with github ml file . There was a mistake , i changed it

**Reply by otseng** [reply]
Can you try replacing this file?
[MaterialMainWindow.zip](https://github.com/eliranwong/UniqueBible/files/9064027/MaterialMainWindow.zip)

**Reply by onegladiator4u** [reply]
yes bro, it works now. Thank you.

**Comment by onegladiator4u** [comment]
Hey guys right now my `widget foreground color` and `dark theme text color` is somewhat same. So verse number and verse shows as same color. How can i change verse color to a different one by css? Is it possible ? Now widget color applies to verse number too.
![h2s2dlQlBd](https://user-images.githubusercontent.com/48617390/180972433-961ed4c3-9f32-4b71-b8d9-9210184ddd98.png)

**Reply by otseng** [reply]
Try dropping this into plugins/event and then modify line 5 to the color you want to change the verse reference color.
[post_parse_bible.zip](https://github.com/eliranwong/UniqueBible/files/9189916/post_parse_bible.zip)

**Reply by onegladiator4u** [reply]
Thank You brother

**Comment by onegladiator4u** [comment]
Hey @eliranwong 
In config flags enableplugins default value is true. That means plugins should automatically show right ? But it's not showing . But when we enable it, it shows. Looks like its vise versa. Or am i wrong ?
![image](https://user-images.githubusercontent.com/48617390/221407809-7dcd50d5-3237-4ef9-9e86-5e9704193009.png)
![image](https://user-images.githubusercontent.com/48617390/221407874-340cab92-666b-491d-99cf-ef6d7d8c64f6.png)

**Reply by otseng** [reply]
For a new installation, the enable plugins is true.  You can test it by exiting UBA, deleting config.py, then restarting UBA.

**Reply by onegladiator4u** [reply]
@otseng Oh i see. Ok brother thanks for clarification.

**Reply by eliranwong** [reply]
> Hey @eliranwong
> In config flags enableplugins default value is true. That means plugins should automatically show right ? But it's not showing . But when we enable it, it shows. Looks like its vise versa. Or am i wrong ?
> ![image](https://user-images.githubusercontent.com/48617390/221407809-7dcd50d5-3237-4ef9-9e86-5e9704193009.png)
> ![image](https://user-images.githubusercontent.com/48617390/221407874-340cab92-666b-491d-99cf-ef6d7d8c64f6.png)

It is by default True, but users can still turn it off.  From your screenshot, I can see it was turned off.

**Reply by onegladiator4u** [reply]
@eliranwong Oh ok. I was actually wondering why it was not showing when it is by default true. I thought true will show it. Was confused earlier.
